### PR TITLE
Multiple changes to remove code

### PR DESCRIPTION
**Pull request by josuah**
I did add some code to jj, so I am in debt to remove some.  :)

I did try to make the commit so that they can be merged or rejected individually, in case only some suits you (variously opinionated changes).

### CODE REVIEW

**Comment by aaronNGi** [comment]
Nice. For now I will have a quick look and leave some comments on the individual changes.

File: jjc [hunk 1]
```diff
@@ -359,8 +355,7 @@ function nick_in_chan(nick, chan)
 	# array are surrounded by spaces, we will always be able to match
 	# whole nicks, instead of getting a partial match when searching
 	# for "john" while that channel also has a user called "johndoe".
-	return index(channels[irc_tolower(chan)],
-		" " irc_tolower(nick) " ")
+	return index(channels[irc_tolower(chan)], " " irc_tolower(nick) " ")
```
Comment (aaronNGi) [inline jjc:358]: I use 74 columns max for the entire project.

File: jjc [hunk 2]
```diff
@@ -69,14 +69,12 @@ BEGIN {
...
-	# BEGIN will have global scope, an array variable is used.  That
-	# way it can easily be emptied later.
-	local["path"]=shell_escape(irc["logdir"] "/channels")
+	# Because all variables used in BEGIN will have global scope, an array
+	# variable is used.  That way it can easily be emptied later.
 
 	# Try to create the "channels" directory while also creating all
 	# parent directories as needed.
-	local["retval"]=system("mkdir -p -- " local["path"] " >&2")
+	local["retval"]=system("mkdir -p -- \"$IRC_DIR/$IRC_HOST/channels\" >&2")
```
Comment (aaronNGi) [inline jjc]: Doesn't fit within 74 columns. We could use:

    local["retval"]=
    	system("mkdir -p -- \"$IRC_DIR/$IRC_HOST/channels\" >&2")

File: jjc [hunk 3]
```diff
@@ -74,11 +74,11 @@ BEGIN {
 
 	# Try to create the "channels" directory while also creating all
 	# parent directories as needed.
-	local["retval"]=system("mkdir -p -- \"$IRC_DIR/$IRC_HOST/channels\" >&2")
+	local["retval"]=system("exec mkdir -p -- \"$IRC_DIR/$IRC_HOST/channels\" >&2")
```
Comment (aaronNGi) [inline jjc]: Exceeds 74 columns. The additional `exec` also makes it too long even when putting the `system()` call on the next line. We could use a variable for the command instead.

File: jjc (hunk 2 above)
Comment (josuah) [reply jjc]: I think there have not really been a coding style for awk applied ever, so it is up to anyone to start one I guess...

**Pull request merged**